mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -664,56 +692,66 @@ abstract class BaseMvRxViewModel<S : MvRxState>(
-            .resolveSubscription(lifecycleOwner, deliveryMode, subscriber)
+        return resolveSubscription(lifecycleOwner, deliveryMode, subscriber)
+            .toDisposable()
             .disposeOnClear()
     }
 
-    private fun <T : Any> Observable<T>.resolveSubscription(
+    private fun <T : Any> Flow<T>.resolveSubscription(
         lifecycleOwner: LifecycleOwner? = null,
         deliveryMode: DeliveryMode,
         subscriber: (T) -> Unit
-    ): Disposable = if (lifecycleOwner == null || FORCE_DISABLE_LIFECYCLE_AWARE_OBSERVER) {
-        this.subscribe(subscriber)
-    } else {
-        this.subscribeWith(
-            MvRxLifecycleAwareObserver(
-                lifecycleOwner,
-                deliveryMode = deliveryMode,
-                lastDeliveredValue = if (deliveryMode is UniqueOnly) {
-                    if (activeSubscriptions.contains(deliveryMode.subscriptionId)) {
-                        throw IllegalStateException(
-                            "Subscribing with a duplicate subscription id: ${deliveryMode.subscriptionId}. " +
-                                "If you have multiple uniqueOnly subscriptions in a MvRx view that listen to the same properties " +
-                                "you must use a custom subscription id. If you are using a custom MvRxView, make sure you are using the proper" +
-                                "lifecycle owner. See BaseMvRxFragment for an example."
-                        )
-                    }
-                    activeSubscriptions.add(deliveryMode.subscriptionId)
-                    lastDeliveredStates[deliveryMode.subscriptionId] as? T
-                } else {
-                    null
-                },
-                onNext = Consumer { value ->
-                    if (deliveryMode is UniqueOnly) {
-                        lastDeliveredStates[deliveryMode.subscriptionId] = value
-                    }
-                    subscriber(value)
-                },
-                onDispose = {
-                    if (deliveryMode is UniqueOnly) {
-                        activeSubscriptions.remove(deliveryMode.subscriptionId)
-                    }
-                }
-            )
-        )
+    ): Job {
+        val flow = if (lifecycleOwner == null || FORCE_DISABLE_LIFECYCLE_AWARE_OBSERVER) {
+            this
+        } else if (deliveryMode is UniqueOnly) {
+            assertOneActiveSubscription(lifecycleOwner, deliveryMode)
+            val lastDeliveredValue: T? = lastDeliveredValue(deliveryMode)
+            this
+                .dropWhile { it == lastDeliveredValue }
+                .flowWhenStarted(lifecycleOwner)
+                .distinctUntilChanged()
+                .onEach { lastDeliveredStates[deliveryMode.subscriptionId] = it }
+        } else {
+            flowWhenStarted(lifecycleOwner)
+        }
+        return flow
+            .onEach { subscriber(it) }
+            .launchIn(viewModelScope)
```

Comment:
Doesn't this leak the fragment? This flow is going to be retained for the life of the view model but references a subscriber (fragment) with a smaller lifecycle. Should this be `lifecycleOwner.lifecycleScope`?